src/runtime_src/core/common/xrt_sensor.cpp

```diff
@@ -0,0 +1,245 @@
+/**
+ * Copyright (C) 2021 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+#define XRT_CORE_COMMON_SOURCE
+
+// Local - Include Files
+#include "xrt_sensor.h"
+#include "query_requests.h"
+#include "utils.h"
+
+// 3rd Party Library - Include Files
+// System - Include Files
+#include <sstream>
+#include <iomanip>
+
+// ------ S T A T I C   F U N C T I O N S -------------------------------------
+namespace {
+template <typename QRVoltage, typename QRCurrent>
```

Comment:
It would be great with a comment as to what this function does and what the template parameters and function arguments are.